### PR TITLE
Feature/upload puzzle image

### PR DESCRIPTION
**Pull request by JRicardoT**
**What (if any) features are you implementing?**
- User is now able to upload a picture of their puzzle in the form.

**What (if anything) did you refactor?**
- Added functions inside the context file to handle the new puzzle being added in.
- Refactored the function being ran after a user submits their puzzle.

**Were there any issues that arose?**
- N/A

**Is there anything needs to be prioritized?**
- Finish testing for this form.

**Next Steps?**
- Go back to cypress branch and finish testing the form.

### CODE REVIEW

File: src/components/AddPuzzleForm.tsx [hunk 1]
```diff
@@ -29,45 +33,64 @@ const AddPuzzleForm = () => {
...
-        id: Date.now().toString(),
-        category: category,
-        missingPieces: missingPieceCount,
-        pieceCount: pieceCount,
-        quality: quality,
-        availability: true,
-        price: price,
-        image: 'https://img.buzzfeed.com/buzzfeed-static/static/2020-04/28/14/asset/1593bcadc012/sub-buzz-488-1588084568-26.jpg'
-      }
+    if(category  && missingPieceCount  && price  && quality  && pieceCount && image) {
```
Comment (ccollums) [inline src/components/AddPuzzleForm.tsx:44]: I think it was a good idea to separate all of this out into their own functions!

File: src/components/AddPuzzleForm.tsx [hunk 2]
```diff
@@ -89,11 +113,26 @@ const AddPuzzleForm = () => {
     setOpenSuccessMessage(false);
   };
 
+  const handleImage = async (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (ccollums) [inline src/components/AddPuzzleForm.tsx:116]: This is amazing! thank you for figuring out how to do this! What a powerful tool.

**Review by ccollums — COMMENTED** [review]
Thank you so much for taking charge on this form! Love the snack bars and ability to upload an image from a user's computer.

File: src/Context.tsx [hunk 3]
```diff
@@ -49,26 +50,21 @@ const PuzzleProvider = ({children}: IPuzzleProvider) => {
...
-	// })
-
 	useEffect(() => {
 		fetchPuzzles();
-	}, [])
+	}, [newPuzzle])
  
 	return (
-		<PuzzleContext.Provider value={{ puzzles, loggedIn, logIn, user, addPuzzle }}>
+		<PuzzleContext.Provider value={{ puzzles, loggedIn, logIn, user, userID, addPuzzle}}>
```
Comment (kbergsund) [inline src/Context.tsx:67]: Let's refactor to remove userID and just use user.id that we already have global access to.

File: src/components/AddPuzzleForm.tsx [hunk 4]
```diff
@@ -14,8 +14,12 @@ const Alert = React.forwardRef<HTMLDivElement, AlertProps>(function Alert(
   return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
 });
 
+const Input = styled('input')({
```
Comment (kbergsund) [inline src/components/AddPuzzleForm.tsx:17]: Can we move this to the .css file or does it need to be inline?

File: src/components/AddPuzzleForm.tsx [hunk 5]
```diff
@@ -89,11 +113,26 @@ const AddPuzzleForm = () => {
     setOpenSuccessMessage(false);
   };
 
+  const handleImage = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files!;
+    const data = new FormData();
+    data.append('file', files[0]);
+    data.append('upload_preset', 'puzzlrs');
+    const res = await fetch('https://api.cloudinary.com/v1_1/dqgqw1dld/image/upload', {
```
Comment (kbergsund) [inline src/components/AddPuzzleForm.tsx:121]: I love cloudinary.

File: src/components/AddPuzzleForm.tsx [hunk 6]
```diff
@@ -165,22 +205,26 @@ const AddPuzzleForm = () => {
...
-            value={missingPieceCount}
-            onChange={event => setMissingPieceCount(event.target.value)}
-          >
-            {piecesOptions}
-          </Select>
-        </FormControl> */}
+        <label htmlFor='upload-photo'>
+          <Input accept="image/*" id="upload-photo" type="file" onChange={e => handleImage(e)}/>
+          <Button
+            sx={{color: 'white', backgroundColor: '#5D736B', '&:hover': {backgroundColor: '#474E4A'}}}
```
Comment (kbergsund) [inline src/components/AddPuzzleForm.tsx:211]: Can we also move this to the .css file?

**Review by kbergsund — APPROVED** [review]
Thorough code review conducted over Zoom together. This is amazing and adds great depth to the user experience of uploading a puzzle!

**Pull request merged**